core/src/main/java/com/google/googlejavaformat/java/java21/Java21InputAstVisitor.java

```diff
@@ -0,0 +1,76 @@
+package com.google.googlejavaformat.java.java21;
+
+import static com.google.common.collect.Iterables.getOnlyElement;
+
+import com.google.googlejavaformat.OpsBuilder;
+import com.google.googlejavaformat.java.java17.Java17InputAstVisitor;
+import com.sun.source.tree.*;
+import java.util.List;
+
+public class Java21InputAstVisitor extends Java17InputAstVisitor {
+
+  public Java21InputAstVisitor(OpsBuilder builder, int indentMultiplier) {
+    super(builder, indentMultiplier);
+  }
+
+  @Override
+  public Void visitCase(CaseTree node, Void unused) {
+    sync(node);
+    markForPartialFormat();
+    builder.forcedBreak();
+    List<? extends CaseLabelTree> labels = node.getLabels();
+    boolean isDefault =
+        labels.size() == 1 && getOnlyElement(labels).getKind().name().equals("DEFAULT_CASE_LABEL");
+    if (isDefault) {
+      token("default", plusTwo);
+    } else {
+      token("case", plusTwo);
+      builder.open(labels.size() > 1 ? plusFour : ZERO);
+      builder.space();
+      boolean first = true;
+      for (Tree expression : labels) {
+        if (!first) {
+          token(",");
+          builder.breakOp(" ");
+        }
+        scan(expression, null);
+        first = false;
+      }
+      builder.close();
+    }
+    if (node.getGuard() != null) {
```

Comment:
Since this block is the only part that needs to be different between 17 and 21, I would like to avoid duplicating the rest of the method.

An alternative would be to do something like this: https://github.com/google/google-java-format/blob/205b85f2315c48b2d6280a18c6ea9dbe814d3709/core/src/main/java/com/google/googlejavaformat/java/JavaInputAstVisitor.java#L3781

There could be a `getGuard` method that returns null in the base class, and `Java21InputAstVisitor` could override it